schema/pgsql/schema.sql

```diff
@@ -123,51 +123,43 @@ CREATE TABLE source (
 );
 
 CREATE TABLE object (
-    id bytea NOT NULL, -- SHA256 of identifying tags
+    id bytea NOT NULL, -- SHA256 of identifying tags and the source.id
+    source_id bigint NOT NULL REFERENCES source(id),
+    name text NOT NULL,
     -- this will probably become more flexible in the future
     host text NOT NULL,
     service text,
 
-    CHECK (length(id) = 256/8),
-
-    CONSTRAINT pk_object PRIMARY KEY (id)
-);
-
-CREATE TABLE source_object (
-    object_id bytea NOT NULL REFERENCES object(id),
-    source_id bigint NOT NULL REFERENCES source(id),
-    name text NOT NULL,
     url text,
 
-    CONSTRAINT pk_source_object PRIMARY KEY (object_id, source_id)
-);
+    CHECK (length(id) = 256/8),
 
+    CONSTRAINT pk_object PRIMARY KEY (id),
+    FOREIGN KEY (source_id) REFERENCES source(id)
+);
 
 CREATE TABLE object_extra_tag (
     object_id bytea NOT NULL REFERENCES object(id),
-    source_id bigint NOT NULL REFERENCES source(id),
-
     tag text NOT NULL,
     value text NOT NULL,
 
-    CONSTRAINT pk_object_extra_tag PRIMARY KEY (object_id, source_id, tag),
-    FOREIGN KEY (object_id, source_id) REFERENCES source_object(object_id, source_id)
+    CONSTRAINT pk_object_extra_tag PRIMARY KEY (object_id, tag),
+    FOREIGN KEY (object_id) REFERENCES object(id)
```

Comment:
That `FOREIGN KEY` definition probably is redundant (should already be created by the `REFERENCES object(id)` in the column defintion).